### PR TITLE
C++: Add more FPs for `cpp/invalid-pointer-deref`

### PR DESCRIPTION
**Pull request by MathiasVP**
Once again: I have no idea why we're getting these FPs when very similar cases are properly handled further up in the file 😂. But I'll leave fixing (and even _understanding_) them to a future PR.

### CODE REVIEW

File: cpp/ql/test/experimental/query-tests/Security/CWE/CWE-193/pointer-deref/test.cpp [hunk 1]
```diff
@@ -254,3 +254,49 @@ void test17(unsigned *p, unsigned x, unsigned k) {
...
     }
 }
+
+struct array_with_size
+{
+  int *xs;
+  unsigned len;
+};
+
+void test17(unsigned len, array_with_size *s)
```
Comment (jketema) [inline cpp/ql/test/experimental/query-tests/Security/CWE/CWE-193/pointer-deref/test.cpp:264]: The `s` argument does not seem to be used and similar for the other functions below.

Comment (MathiasVP) [reply cpp/ql/test/experimental/query-tests/Security/CWE/CWE-193/pointer-deref/test.cpp]: Thanks. I managed to reduce the issue even further, but forgot to remove the (now superfluous) parameter.

**Review by jketema — APPROVED** [review]
LGTM

**Comment by MathiasVP** [comment]
🤠 merging!

**Pull request merged**